api/preprints/views.py

```diff
@@ -350,3 +354,72 @@ def get_object(self):
             return {'citation': citation, 'id': style}
 
         raise PermissionDenied if auth.user else NotAuthenticated
+
+
+class PreprintIdentifierList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin, PreprintMixin):
```

Comment:
Hm, the inheritance doesn't feel quite right--I would expect this to inherit from `IdentifierList` (you'll need to generalize IdentifierList a bit to make this work).

```python
# api/identifiers/views.py
class IdentifierList(...):
    def get_object(self...):
        raise NotImplementedError

# api/preprints/views.py
class PreprintIdentifierList(..., IdentifierList, PreprintMixin):
     # Override IdentifierList
     def get_object(self, check_object_permissions=True):
         return self.get_preprint(check_object_permissions=check_object_permissions)

# do something simpliar with NodeIdentifierList
```